src/main/java/at/hannibal2/skyhanni/features/mining/crystalhollows/CrystalHollowsProfessorRobot.kt

```diff
@@ -0,0 +1,52 @@
+package at.hannibal2.skyhanni.features.mining.crystalhollows
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.data.IslandType
+import at.hannibal2.skyhanni.events.LorenzChatEvent
+import at.hannibal2.skyhanni.utils.ChatUtils
+import at.hannibal2.skyhanni.utils.HypixelCommands
+import at.hannibal2.skyhanni.utils.InventoryUtils
+import at.hannibal2.skyhanni.utils.ItemUtils.name
+import at.hannibal2.skyhanni.utils.LorenzUtils.isInIsland
+import at.hannibal2.skyhanni.utils.StringUtils.removeColor
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+class CrystalHollowsProfessorRobot {
+
+    private val config get() = SkyHanniMod.feature.mining.professorRobot;
+
+    private val pattern = "\\[NPC\\] Professor Robot: That's not one of the components I need! Bring me one of the missing components:".toRegex()
+
+    private var robotMessage = false
+    private val robotParts = listOf(
+        "Electron Transmitter",
+        "FTX 3070",
+        "Robotron Reflector",
+        "Superlite Motor",
+        "Control Switch",
+        "Synthetic Heart"
+    )
+
+    @SubscribeEvent
+    fun onChat(event: LorenzChatEvent) {
+        if (!isEnabled()) return
+        if (event.message.removeColor().matches(pattern)) {
```

Comment:
save the removeColor here. 
```suggestion
       val cleanMessage = event.message.removeColor()
        if (cleanMessage matches(pattern)) {
```